app/gui/src/dashboard/modals/CreateCredentialModal.tsx

```diff
@@ -0,0 +1,45 @@
+/** @file A modal for creating and editing a credential. */
+import { Dialog, Dropdown, Text } from '#/components/AriaComponents'
+import { CREDENTIAL_INFOS } from '#/data/serviceCredentials'
+import { makeCredentialCreationHandler } from '#/data/serviceCredentials/logic'
+import { useText } from '#/providers/TextProvider'
+import type { CredentialConfig, SecretId } from '#/services/Backend'
+import { useState } from 'react'
+
+/** Props for a {@link CreateCredentialModal}. */
+export interface CreateCredentialModalProps {
+  readonly noDialog?: boolean
+  readonly doCreate: (name: string, value: CredentialConfig) => Promise<SecretId>
+}
+
+/** A modal for creating a credential. */
+export default function CreateCredentialModal(props: CreateCredentialModalProps) {
+  const { noDialog = false, doCreate } = props
```

Comment:
noDialog should be forbidden, prefer exporting 2 components: `CreateCredentialModal ` and `CreateCredentialForm`